docs/layouts/shortcodes/generated/expert_forst_section.html

```diff
@@ -0,0 +1,54 @@
+    <thead>
+        <tr>
+            <th class="text-left" style="width: 20%">Key</th>
+            <th class="text-left" style="width: 15%">Default</th>
+            <th class="text-left" style="width: 10%">Type</th>
+            <th class="text-left" style="width: 55%">Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><h5>state.backend.forst.executor.inline-coordinator</h5></td>
+            <td style="word-wrap: break-word;">false</td>
+            <td>Boolean</td>
+            <td>Whether to let the task thread be the coordinator thread responsible for distributing requests.</td>
+        </tr>
+        <tr>
+            <td><h5>state.backend.forst.executor.inline-write</h5></td>
+            <td style="word-wrap: break-word;">true</td>
+            <td>Boolean</td>
+            <td>Whether to let write requests executed within the coordinator thread.</td>
+        </tr>
+        <tr>
+            <td><h5>state.backend.forst.local-dir</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>String</td>
+            <td>The local directory (on the TaskManager) where ForSt puts some meta files. Per default, it will be &lt;WORKING_DIR&gt;/tmp. See <code class="highlighter-rouge">process.taskmanager.working-dir</code> for more details.</td>
+        </tr>
+        <tr>
+            <td><h5>state.backend.forst.memory.fixed-per-slot</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>MemorySize</td>
+            <td>The fixed total amount of memory, shared among all ForSt instances per slot. This option overrides the 'state.backend.forst.memory.managed' option when configured.</td>
+        </tr>
+        <tr>
+            <td><h5>state.backend.forst.memory.fixed-per-tm</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>MemorySize</td>
+            <td>The fixed total amount of memory, shared among all ForSt instances per Task Manager (cluster-level option). This option only takes effect if neither 'state.backend.forst.memory.managed' nor 'state.backend.forst.memory.fixed-per-slot' are not configured. If none is configured then each ForSt column family state has its own memory caches (as controlled by the column family options). The relevant options for the shared resources (e.g. write-buffer-ratio) can be set on the same level (config.yaml).Note, that this feature breaks resource isolation between the slots</td>
+        </tr>
+        <tr>
+            <td><h5>state.backend.forst.memory.managed</h5></td>
+            <td style="word-wrap: break-word;">true</td>
+            <td>Boolean</td>
+            <td>If set, the ForSt state backend will automatically configure itself to use the managed memory budget of the task slot, and divide the memory over write buffers, indexes, block caches, etc. That way, the three major uses of memory of ForSt will be capped.</td>
```

Comment:
we say` the three major uses of memory` but we list `write buffers, indexes, block caches, etc` i.e. more than 3. 

I suggest we document when not set all the options the user needs to provide.

I suggest including guidance as to when to et this and when not to.

Reply:
This is the doc generated from the code. We will have another guidance doc if needed.

The document you are reading is 'expert' section. Typically we don't expect user to tuning these options (and they sometimes are not recommended for some reason). They should know about the memory management of Flink in detail first.

Reply:
I removed the last sentence, since it's misleading.